HW 30/main.js

```diff
@@ -0,0 +1,154 @@
+// Спливання та перехоплення дозволяють нам реалізувати один з найпотужніших шаблонів обробки подій під назвою делегування подій.
+// Ідея в тому, що якщо у нас є багато елементів, які обробляються подібним чином, то замість того, щоб призначати обробник кожному з них,
+// ми ставимо один обробник на їхнього спільного предка.
+// У обробнику ми отримуємо event.target, щоб побачити, де насправді сталася подія і обробити її.
+
+let ul = document.querySelector('#ul');
+
+ul.addEventListener('mousedown', function (e) {
+    e.preventDefault();
+});
+
+ul.addEventListener('click', function (e) { 
+    if (e.target == this) {
+        return false;
+    };
+
+    if (e.shiftKey) {
+        chooseSelected(this.children, e.target, ul);
+    } 
+    else {
+        if (!e.ctrlKey) {
+            clearSelected(this.children);
+        };   
+        toggleSelected(e.target);
+    }; 
+});
+
+function clearSelected(elems) {
+    for (let elem of elems) {
+        elem.classList.remove('selected');
+    };
+};
+
+function chooseSelected(elems, target, e) { 
+    target.classList.toggle('selected'); 
+    let node = target.previousElementSibling  
+
+    while(node != null) { 
+        if (node.classList != 'selected') {
+            node.classList.toggle('selected');
+        }
+        else {
+            break
+        };
```

Comment:
If we have the opportunity to not use `else`  we need to do that. Change condition to use break inside `if` then you'll get rid of `else`